src/main/kotlin/creator/custom/model/TemplateApi.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Minecraft Development for IntelliJ
+ *
+ * https://mcdev.io/
+ *
+ * Copyright (C) 2024 minecraft-dev
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, version 3.0 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.demonwav.mcdev.creator.custom.model
+
+/**
+ * Marker annotation indicating classes exposed to templates.
+ *
+ * Be careful of not breaking source or binary compatibility of those APIs without a good reason.
+ */
+annotation class TemplateApi
```

Comment:
Is this enforced in any way or just documentation? Also, it should probably be `@Target(CLASS)` and with an appropriate `@Retention`.

Reply:
This is only meant for documentation. Will add Target and Retention 👌